cookbook/controller/argument_value_resolver.rst

```diff
@@ -0,0 +1,230 @@
+calls ``resolve()`` with the same values as it called ``supports()``.
+
+That's it! Now all you have to do is add the configuration for the service
+container. This can be done by tagging the service with ``kernel.argument_resolver``
+and adding a priority.
+
+.. note::
+
+    While adding a priority is optional, it's recommended to add one to
+    make sure the expected value is injected. The ``ArgumentFromAttributeResolver``
+    has a priority of 100. As this  one is responsible for fetching attributes
+    from the ``Request``, it's also recommended to trigger your custom value
+    resolver with a lower priority. This makes sure the argument resolvers
+    are not triggered in (e.g.) subrequests if you pass your user along:
+    ``{{ render(controller('AppBundle:User:index', {'user', app.user})) }}``.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            app.value_resolver.user:
+                class: AppBundle\ArgumentValueResolver\UserValueResolver
+                arguments:
+                    - '@security.token_storage'
+                tags:
+                    - { name: controller.argument_value_resolver, priority: 50 }
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="'http://www.w3.org/2001/XMLSchema-Instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="app.value_resolver.user" class="AppBundle\ArgumentValueResolver\UserValueResolver">
+                    <argument type="service" id="security.token_storage">
+                    <tag name="controller.argument_value_resolver" priority="50" />
+                </service>
+            </services>
+
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use Symfony\Component\DependencyInjection\Definition;
+
+        $defintion = new Definition(
+            'AppBundle\ArgumentValueResolver\UserValueResolver',
+            array(new Reference('security.token_storage'))
+        );
+        $definition->addTag('controller.argument_value_resolver', array('priority' => 50));
+        $container->setDefinition('app.value_resolver.user', $definition);
+
+.. _`yield`: http://php.net/manual/en/language.generators.syntax.php
+
+Creating an Optional User Resolver
+----------------------------------
+
+When you want your user to be optional, e.g. when your page is behind a
+firewall that also allows anonymous authentication, you might not always
+have a security user. To get this to work, you only have to change your
+method signature to `UserInterface $user = null`.
+
+When you take the ``UserValueResolver`` from the previous example, you can
```

Comment:
Ok sorry, I missed the `ArgumentResolver` exception. Maybe it could be enough to explicit it somewhere in this PR?

For the naming I used the same for formatters' interface in https://github.com/symfony/symfony/pull/18450 but I was just wondering why this convention of an "s" for `supports()`, maybe because it's passive (contrary to an action of supporting)?